pyscraper/new_hansard.py

```diff
@@ -0,0 +1,1442 @@
+            elif re.match("Six o'clock(?i)", time):
+                    newtime = '18:00:00'
+
+            self.time = newtime
+        elif re.match('.*was asked.*', text):
+            self.parse_para_with_member(date, None)
+
+    def parse_oral_heading(self, heading):
+        self.output_heading = True
+        tag = etree.Element('oral-heading')
+        tag.set('id', self.get_speech_id())
+        tag.set('nospeaker', 'true')
+        tag.set('colnum', self.current_col)
+        tag.set('time', self.current_time)
+        tag.set(
+            'url',
+            'http://www.publications.parliament.uk{0}'.format(
+                heading.get('url')
+            )
+        )
+        tag.text = heading.text
+        self.root.append(tag)
+
+    def parse_major(self, heading, **kwargs):
+        self.check_for_pi(heading)
+        text = self.get_text_from_element(heading)
+        if text.strip() == 'Prayers':
+            return
+        self.clear_current_speech()
+        tag = etree.Element('major-heading')
+        if self.uc_titles:
+            tag.text = text.upper()
+        else:
+            tag.text = text
+
+        if 'extra_text' in kwargs:
+            tag.text = u'{0} - '.format(tag.text)
+            i = etree.Element('i')
+            i.text = kwargs['extra_text']
+            tag.append(i)
+
+        tag.set('id', self.get_speech_id())
+        tag.set('nospeaker', 'true')
+        tag.set('colnum', self.current_col)
+        tag.set('time', self.current_time)
+        tag.set(
+            'url',
+            'http://www.publications.parliament.uk{0}'.format(
+                heading.get('url')
+            )
+        )
+        self.root.append(tag)
+        self.output_heading = True
+        self.current_speech_num = self.current_speech_num + 1
+
+    def parse_minor(self, heading):
+        self.check_for_pi(heading)
+        if self.current_speech is not None:
+            self.clear_current_speech()
+        tag = etree.Element('minor-heading')
+        tag.set('id', self.get_speech_id())
+        tag.set('nospeaker', 'true')
+        tag.set('colnum', self.current_col)
+        tag.set('time', self.current_time)
+        tag.set(
+            'url',
+            'http://www.publications.parliament.uk{0}'.format(
+                heading.get('url')
+            )
+        )
+        text = self.get_text_from_element(heading)
+        tag.text = text
+        self.root.append(tag)
+        self.output_heading = True
+        self.current_speech_num = self.current_speech_num + 1
+
+    def parse_generic(self, heading):
+        self.check_for_pi(heading)
+        if self.current_speech_num <= 0:
+            self.parse_major(heading)
+        else:
+            self.parse_minor(heading)
+
+    def parse_opposition(self, heading):
+        following = heading.xpath(
+            '(./following-sibling::ns:hs_2cDebatedMotion|./following-sibling::ns:hs_7SmCapsHdg|./following-sibling::ns:hs_2GenericHdg)',
```

Comment:
Yesterday's following heading was hs_2DebatedMotion. But see my comment elsewhere on that class, as to whether that should be a minor heading or changed somehow. There are sometimes too backbench debates, such as on http://www.theyworkforyou.com/debates/?d=2016-01-27 – but if there's only one, I guess we'd like it smooshed together, but meh.